### PR TITLE
Data picker - special case the `children` prop

### PR DESCRIPTION
**Pull request by bkrmendy**
# Try it [here](https://utopia.fish/p/cc22c5f3-charming-saturnalia/?branch_name=feature-children-prop-dedicated-picking)

## Problem
When the `children` prop is requested via `registerModule`/`registerInternalComponent`, and the data picker is being used to pick a variable for the `children` prop, it puts the picked variable in the wrong place.

For example, consider this component:

```tsx
<NavLink key={item.id} prefetch="intent" style={activeLinkStyle} to={url}>
  <p>Children should go here</p>
</NavLink>
```

When picking a variable to replace the `p` inside `Navlink`, the result becomes this:
```tsx
<NavLink key={item.id} prefetch="intent" style={activeLinkStyle} to={url} children={item.title}>
  <p>Children should go here</p>
</NavLink>
```

instead of:

```tsx
<NavLink key={item.id} prefetch="intent" style={activeLinkStyle} to={url}>
  {item.title}
</NavLink>
```

## Fix
Special case the `children` prop, and instead of setting a prop on the selected element, put the selected variable into the slot for `children`.

### CODE REVIEW

File: editor/src/components/editor/actions/actions.tsx [hunk 1]
```diff
@@ -2231,6 +2232,30 @@ export const UPDATE_FNS = {
...
   },
+  INSERT_ATTRIBUTE_OTHER_JAVASCRIPT_INTO_ELEMENT: (
+    action: InsertAttributeOtherJavascriptIntoElement,
+    editor: EditorModel,
+  ): EditorModel => {
+    const withNewElement = modifyUnderlyingTargetElement(action.parent, editor, (element) => {
+      switch (element.type) {
+        case 'JSX_CONDITIONAL_EXPRESSION':
+        case 'JSX_FRAGMENT':
+          // only JSX elements are handled in this action
```
Comment (balazsbajorics) [inline editor/src/components/editor/actions/actions.tsx:2243]: should this throw an Error?

File: editor/src/components/inspector/sections/component-section/data-picker-popup.tsx [hunk 2]
```diff
@@ -70,6 +70,14 @@ function valueToDisplay(option: VariableOption): string {
   }
 }
 
+function isChildrenProp(path: PropertyPath): boolean {
+  return (
+    path.propertyElements.length > 0 &&
```
Comment (balazsbajorics) [inline editor/src/components/inspector/sections/component-section/data-picker-popup.tsx:75]: what if the path is longer than 1 element? it feels like we should error out in that case (not necessarily here, but this code reminded me of the problem)

**Review by balazsbajorics — APPROVED** [review]
looks good, left some minor questions

File: editor/src/components/inspector/sections/component-section/data-picker-popup.tsx (hunk 2 above)
Comment (bkrmendy) [reply editor/src/components/inspector/sections/component-section/data-picker-popup.tsx:75]: fair concern, I'll address this in follow-up PR so the decision around this edge isn't muddled together with the code on this PR

File: editor/src/components/editor/actions/actions.tsx [hunk 3]
```diff
@@ -2231,6 +2232,30 @@ export const UPDATE_FNS = {
       selectedViews: newSelectedViews,
     }
   },
+  INSERT_ATTRIBUTE_OTHER_JAVASCRIPT_INTO_ELEMENT: (
```
[the remark this answers is not in the document]
Comment (bkrmendy) [reply editor/src/components/editor/actions/actions.tsx:2235]: how about `INSERT_JS_INTO_JSX_CHILDREN`? Trying to balance name length vs information density here :D

**Review by seanparsons — APPROVED** [review]

**Pull request merged**